dao/src/main/java/se/fortnox/reactivewizard/db/transactions/StatementConnectionScheduler.java

```diff
@@ -0,0 +1,7 @@
+package se.fortnox.reactivewizard.db.transactions;
+
+import rx.Scheduler;
+import se.fortnox.reactivewizard.db.ConnectionProvider;
+import se.fortnox.reactivewizard.db.statement.Statement;
+
+public record StatementConnectionScheduler(Statement statement, ConnectionProvider connectionProvider, Scheduler scheduler) { }
```

Comment:
I think that the scheduler and connection can be bundled tighter. Something that will both provide a connection and a correct thread, like this:
```
connectionScheduler.run(connection->{
   for (Statement s : statements) {
      s.run(subscription, connection);
   }
})
```
It would give fewer moving parts.

We will eventually need to package the `onTransactionCompleted` inside a wrapping class as well, so it would be nice to bundle that as well in here, so that DaoObservable will wrap a single state.

Naming stuff is not my speciality. It is basically a Statement and the context needed to execute the statement in a transaction. StatementContext?

Reply:
Something like this?